### PR TITLE
2933 Return an alert is a course membership already exists

### LINKED ISSUE #2933: Return an alert if a course membership already exists
If a user tries to create a new course_membership for someone, but that person is already enrolled in the course, we should respond with an alert message: '`First-Name` `Last-Name` is already enrolled in this course site" rather than a server error

### PR DESCRIPTION
**Pull request by dovek**
### Status
**IN DEVELOPMENT**

### Description
If a user tries to create a new course_membership for someone, but that person is already enrolled in the course, we should respond with an alert message: 'First-Name Last-Name is already enrolled in this course site" rather than a server error

### Related PRs
N/A

### Todos
- [ ] Add Tests

### Deploy Notes
N/A

### Gem dependencies
N/A

### Migrations
NO

### Steps to Test or Reproduce
As an admin, navigate to the students page and add a new student that already exists in the course. I should return to the user page with an alert that the user already exists in said course.

However, if I navigate to a second course where the student does not have a course membership, I should still be able to add that student.

### Impacted Areas in Application
* Students

======================
Closes #2933

### CODE REVIEW

**Comment by dovek** [comment]
@chcholman I made some changes in the users_controller that will prevent duplicate users (as well as duplicate course_memberships). I'm not sure if that was in the scope of this specific assignment, or if it was even needed. Feel free to let me know if you think it should be taken out. We can talk about it tomorrow at our check-in if you want.

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -74,26 +74,35 @@ def edit
...
     end
 
-    CourseMembershipBuilder.new(current_user).build_for(@user)
-    render :new
+    if @user.course_memberships.where(course_id: current_course.id).first.nil?
+      CourseMembershipBuilder.new(current_user).build_for(@user)
+      render :new
+    else
+      redirect_to students_path, alert: "#{term_for :student} #{params["user"]["first_name"]}
+        #{params["user"]["last_name"]} with email #{params["user"]["email"]} already exists for #{current_course.name}!"
```
Comment (chcholman) [inline app/controllers/users_controller.rb]: 👍 great job getting the `term_for` in there!

Comment (chcholman) [reply app/controllers/users_controller.rb]: You're checking to see if the course membership exists - but then if it does, you're always assuming it's a student. How could you rewrite it to support users of all roles?

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -248,6 +257,20 @@ def resend_activation_email
 
   private
 
+  def check_params
+    if User.find_by_insensitive_email(params["user"]["email"]).nil?
```
Comment (chcholman) [inline app/controllers/users_controller.rb]: Would it make sense to rename this to something more directly related to what it does? I was thinking along the lines of "user_exists" - thoughts?

File: app/importers/user_importers/csv_student_importer.rb [hunk 3]
```diff
@@ -34,6 +34,11 @@ def import(course=nil)
           next
         end
 
+        if check_user(user, team, course)
+          append_unsuccessful row, "Unable to create course membership for student, course membership already exists"
```
Comment (chcholman) [inline app/importers/user_importers/csv_student_importer.rb]: Let's update this language to be a little bit more oriented to the user: `Unable to import this #{term_for :student}, they have already been added to the course`

File: app/importers/user_importers/csv_student_importer.rb [hunk 4]
```diff
@@ -34,6 +34,11 @@ def import(course=nil)
...
+          next
+        end
+
         unless Services::CreatesCourseMembership.create(user, course).success?
           append_unsuccessful row, "Unable to create course membership for student"
```
Comment (chcholman) [inline app/importers/user_importers/csv_student_importer.rb]: And let's fix this comment while we're at it - `Unable to add this #{term_for :student} to the course`

File: spec/importers/user_importers/csv_student_importer_spec.rb [hunk 5]
```diff
@@ -58,6 +58,15 @@
           expect(team.students.first.email).to eq "csv_jimmy@example.com"
         end
 
+        it "appends unsuccessful if a course membership already exists for user and there is no change to team" do
+          # create :user, first_name: "Jimmy", last_name: "Page", email: "csv_jimmy@example.com",
```
Comment (chcholman) [inline spec/importers/user_importers/csv_student_importer_spec.rb]: Can we remove the commented out lines if they're unnecessary?

File: spec/controllers/users_controller_spec.rb [hunk 6]
```diff
@@ -63,7 +63,59 @@
         post :update, params: { id: user.id, user: params }
         expect(user.reload.first_name).to eq "Jonathan"
       end
+    end
 
+    context "calling create" do
+
```
Comment (chcholman) [inline spec/controllers/users_controller_spec.rb]: 👍

**Pull request merged**